templates/template_db.go

```diff
@@ -0,0 +1,128 @@
+package templates_v1
+
+import (
+	"fmt"
+
+	"context"
+
+	"github.com/jackc/pgx"
+)
+
+func dbConnection() (*pgx.Conn, error) {
+	config := pgx.ConnConfig{
+		Host:     "localhost",
+		Database: "tsg",
+		Port:     26257,
+		User:     "root",
+	}
+
+	return pgx.Connect(config)
+}
+
+func FindTemplateBy(key string) (*MachineTemplate, bool) {
+	var template MachineTemplate
+
+	sqlStatement := `SELECT name, package, image_id FROM triton.tsg_templates WHERE name = $1;`
+
+	db, err := dbConnection()
+	if err != nil {
+		panic(err)
+	}
+
+	defer db.Close()
+
+	err = db.QueryRowEx(context.TODO(), sqlStatement, nil, key).Scan(&template.Name, &template.Package, &template.ImageID)
+	switch err {
+	case nil:
+		return &template, true
+	case pgx.ErrNoRows:
+		fmt.Println("No rows were returned!")
+		return nil, false
+	default:
+		panic(err)
+	}
+}
+
+func FindTemplates() ([]*MachineTemplate, error) {
+	var templates []*MachineTemplate
+
+	sqlStatement := `SELECT name, package, image_id FROM triton.tsg_templates;`
+
+	db, err := dbConnection()
+	if err != nil {
+		return nil, err
+	}
+
+	defer db.Close()
+
+	rows, err := db.QueryEx(context.TODO(), sqlStatement, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var template MachineTemplate
+		err := rows.Scan(&template.Name, &template.Package, &template.ImageID)
```

Comment:
I would move this `Scan()` call to be above so it's chained off of the `QueryEx()` as we did earlier.  We don't loose anything w.r.t. error handling if we do that since `Scan()` will return the error from `QueryEx()` if `QueryEx()` runs into a problem.

Reply:
This is Query not QueryRow so it doesn't have Scan available, it needs to be in a loop

Reply:
Correct, sorry about that.